File: AddIn/DemoAddIn/cs/DemoAddIn/Ribbon2d.cs
using SolidEdgeCommunity.AddIn;
using SolidEdgeCommunity.Extensions; // https://github.com/SolidEdgeCommunity/SolidEdge.Community/wiki/Using-Extension-Methods
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DemoAddIn
{
    class Ribbon2d : SolidEdgeCommunity.AddIn.Ribbon
    {
        const string _embeddedResourceName = "DemoAddIn.Ribbon2d.xml";

        public Ribbon2d()
        {
            // Get a reference to the current assembly. This is where the ribbon XML is embedded.
            var assembly = System.Reflection.Assembly.GetExecutingAssembly();

            // In this example, XML file must have a build action of "Embedded Resource".
            this.LoadXml(assembly, _embeddedResourceName);
        }

        public override void OnControlClick(RibbonControl control)
        {
        }
    }
}

File: AddIn/DemoAddIn/vb/DemoAddIn/Ribbon2d.vb
Imports SolidEdgeCommunity.AddIn
Imports SolidEdgeCommunity.Extensions ' https://github.com/SolidEdgeCommunity/SolidEdge.Community/wiki/Using-Extension-Methods
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Friend Class Ribbon2d
    Inherits SolidEdgeCommunity.AddIn.Ribbon

    Private Const _embeddedResourceName As String = "DemoAddIn.Ribbon2d.xml"

    Public Sub New()
        ' Get a reference to the current assembly. This is where the ribbon XML is embedded.
        Dim assembly = System.Reflection.Assembly.GetExecutingAssembly()

        ' In this example, XML file must have a build action of "Embedded Resource".
        Me.LoadXml(assembly, _embeddedResourceName)
    End Sub

    Public Overrides Sub OnControlClick(ByVal control As RibbonControl)
    End Sub
End Class
